lib/rspec/core/reporter.rb

```diff
@@ -2,6 +2,15 @@ module RSpec::Core
   # A reporter will send notifications to listeners, usually formatters for the
   # spec suite run.
   class Reporter
+    # @private
+    NOTIFICATIONS =
+      [
+        :close, :deprecation, :deprecation_summary, :dump_failures, :dump_pending,
+        :dump_profile, :dump_summary, :example_failed, :example_group_finished,
+        :example_group_started, :example_passed, :example_pending, :example_started,
+        :message, :seed, :start, :start_dump, :stop
+      ]
```

Comment:
>So I still think this should be a set for better perf, given that this isn't a super small array (e.g. <= 5 items).

I didn't want to use `Set` because it requires us to require it, but I've realised we already do for the deprecation formatter... I have mixed feelings about that, I know the performance is better but I'd really like to avoid requiring parts of the standard library, it's something that can catch us out.

> Also, I'd still like something that enforces that we keep this up to date. One idea: make notify verify that the given event is included in this list, unless the caller of notify passes a :skip_supported_notification_verification option (or similar...that's kind a long option name). Then from publish it can pass that flag, but for all other notifications it would only be allowed if the event is in this list, which would enforce that this list is always kept up-to-date with all lifecycle events.

Personally I'd like to avoid the overhead of having to check each time in notify, but I'd be willing to work up a test to check that they're all included somehow?

> Also, should the constant be RESERVED_NOTIFICATIONS or LIFECYCLE_NOTIFICATIONS or something?

I've changed it to `RSPEC_NOTIFICATIONS` if you think that helps?